contrib/k8s.io/gateway-api/cmd/patch-gateways/main.go

```diff
@@ -0,0 +1,231 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"slices"
+	"strconv"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/utils/ptr"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	selector  = flag.String("selector", "", "Label selector to filter Gateways")
+	prompt    = flag.Bool("prompt", false, "Prompt before applying changes to each Gateway")
+	timeout   = flag.Duration("timeout", 1*time.Minute, "Timeout for the operation")
+	namespace = flag.String("namespace", "", "Namespace where the request-mirror is (defaults to current context)")
+)
+
+var gvr = schema.GroupVersionResource{
+	Group:    "gateway.networking.k8s.io",
+	Version:  "v1",
+	Resource: "gateways",
+}
+
+func main() {
+	flag.Parse()
+	log := log.New(os.Stdout, "", 0)
```

Comment:
Logging on STDOUT? 🤨

```suggestion
	log := log.New(os.Stderr, "", 0)
```